museek/plugin/sanity_check_observation_plugin.py

```diff
@@ -220,3 +222,56 @@ def create_plots_of_scan_data(self, data: TimeOrderedData):
         plt.hist(data.elevation.squeeze.flatten(), bins=200)
         plt.xlabel('elevation')
         self.savefig(description='Elevation histogram of all dishes during scan.')
+
+
+    def check_closeness_to_sunrise_sunset(self, data: TimeOrderedData, report_writer: ReportWriter):
+
+        """
+        Check if observation was done within 30 min of sunset or sunrise.
+        :param data: the `TimeOrderedData` to check
+        :param report_writer: the `ReportWriter` object to handle the report
+        """
+
+        import ephem
+        from datetime import datetime, timedelta
+        import numpy as np
+
+        observer = ephem.Observer()
+        observer.lat = '-30.7130'  # Latitude of Meerkat
+        observer.lon = '21.4203'  # Longitude of Meerkat
+
+        # Set the observer's date
+        date_time = datetime.utcfromtimestamp(float(data.original_timestamps[0]))
```

Comment:
I would use `data.timestamp_dates`. Maybe that's already what you want

Reply:
data.timestamp_dates is different with data.original_timestamps, original_timestamps[0] gives the start of the whole scan (including the calibration), timestamp_dates[0] gives the start of the scanning on scan_azel_with_nd_trigger

Reply:
Okay, in this case I would suggest to use the result of the `InPlugin` called `OBSERVATION_DATE`. If it's not exactly what you want, please change it according to your needs (nobody else is using this result yet, but it's purpose is exactly this). You can access it here if you add `ResultEnum.OBSERVATION_DATE` to this plugin's requirements and add a new parameter, e.g. `observation_date` to the run method

Reply:
The OBSERVATION_DATE and ResultEnum.OBSERVATION_DATE is utilzed now.